common/djangoapps/third_party_auth/models.py

```diff
@@ -149,7 +160,7 @@ def clean(self):
     def provider_id(self):
         """ Unique string key identifying this provider. Must be URL and css class friendly. """
         assert self.prefix is not None
-        return "-".join((self.prefix, ) + tuple(getattr(self, field) for field in self.KEY_FIELDS))
+        return "-".join((self.prefix, ) + tuple(str(getattr(self, field)) for field in self.KEY_FIELDS))
```

Comment:
Note that provider_ids are sometimes used in URLs, so changing the provider_id could cause breaking changes.

For example, university partners can currently do a "hinted login" by sending their users to a URL like [https://edge.edx.org/courses/course-v1:UBC+CourseName+2016/courseware/**?tpa_hint=saml-ubc-edge**](https://edge.edx.org/courses/course-v1:UBC+CourseName+2016/courseware/?tpa_hint=saml-ubc-edge) (Open that example in an incognito window to see), where the argument is the `provider_id`. So if you change the format of the `provider_id`, such links would stop working.

My suggestion (see previous comment) is that we don't make `site_id` a key field, and that we give OAuth2 providers a new "slug" field that defaults to the backend_name, which would preserve backward compatibility.